src/material/core/datetime/date-adapter.ts

```diff
@@ -203,6 +203,14 @@ export abstract class DateAdapter<D> {
    */
   abstract invalid(): D;
 
+  /**
+   * @param obj The object to check.
+   * @returns The given object if it is both a date instance and valid, otherwise null.
+   */
+  getValidDateOrNull(obj: any): D | null {
```

Comment:
I believe this should be `unknown` instead of `any`
(`unknown` didn't exist back when we first made the `DateAdapter`)